q2_quality_control/decontam.py

```diff
@@ -0,0 +1,172 @@
+            raise ValueError('Argument to %r was %r, should be %s.'
+                             % (param, arg, explanation))
+
+
+def _check_column_inputs_helper(table, metadata,
+                                prev_control_column,
+                                prev_control_indicator):
+    if prev_control_column not in metadata.columns:
+        raise ValueError('Prevalence column not found, please '
+                         'select from:\n'
+                         + str(', '.join(metadata.columns)))
+    else:
+        if prev_control_indicator not in list(
+                metadata[prev_control_column]):
+            raise ValueError('No control values found, please select '
+                             'from:\n' +
+                             str(', '.join(
+                                 metadata[prev_control_column]
+                                 .unique())))
+        else:
+            prev_controls = metadata.loc[
+                metadata[prev_control_column] == prev_control_indicator]
+            prev_sample_names = prev_controls.index.values
+            indic = 0
+            for name in prev_sample_names:
+                if name in table.index.values:
+                    indic = indic + 1
+            if indic < 5:
+                raise ValueError('At least 5 Control Samples needed '
+                                 + str(indic) + ' found')
+            else:
+                print("All appropriate inputs are found")
+
+
+def _check_column_inputs(table, metadata, method, freq_concentration_column,
+                         prev_control_column, prev_control_indicator):
+    if method == 'prevalence':
+        _check_column_inputs_helper(table, metadata,
+                                    prev_control_column,
+                                    prev_control_indicator)
+
+    elif method == 'frequency':
+        if freq_concentration_column not in metadata.columns:
+            raise ValueError('Frequency column not found, please '
+                             'select from:\n'
+                             + str(', '.join(metadata.columns)))
+        else:
+            print("All appropriate inputs are found")
+    else:
+        if ((freq_concentration_column not in metadata.columns) or
+                (prev_control_column not in metadata.columns)):
+            raise ValueError('Column id input error, please '
+                             'select from:\n'
+                             + str(', '.join(metadata.columns)))
+        else:
+            _check_column_inputs_helper(table, metadata,
+                                        prev_control_column,
+                                        prev_control_indicator)
+
+
+def _decontam_identify_helper(track_fp, method):
+    df = pd.read_csv(track_fp, sep='\t', index_col=0)
+    df.index.name = '#OTU ID'
+    df = df.drop(df.columns[(len(df.columns)-1)], axis=1)
+    if method == 'combined':
+        df = df.fillna(0)
+    df = df.dropna(axis='columns')
```

Comment:
Ok, I think we need to rework how the results are handled. Instead of having `df.dropna` remove columns, we should proactively select the columns we expect from `method`.

Also, it appears that when `prev`  equals `1`, `p` is `NaN`. Should we be filling in with `0` just like in the `"combined"` method?

Reply:
Looks like this is the relevant section: https://github.com/benjjneb/decontam/blob/master/R/decontam.R#L304

I guess a NaN makes sense, since we've only seen that feature once.

So then really what we need to address is the ability to compose the scores qza with the visualizer, which currently expects `p` to exist (and KeyErrors when this drop-NA happens). The `p` column should probably always exist, but if all entries are NaN, it can either produce that as the report or error.

Reply:
hmmm okay how would you like to go about that? I could report which proportion of of the ASVs are NaNs in the output and then raise an error if it is 100%? Also yes you are right we should be filling the NaNs for prev as well which I will change in the next push. Do you think we should report which ASVs are dropped?